plugins/zns/zns.c

```diff
@@ -223,7 +224,26 @@ static int zone_mgmt_send(int argc, char **argv, struct command *cmd, struct plu
 	if (fd < 0)
 		return errno;
 
-	if (cfg.data_len) {
+	if (!cfg.zsa) {
+		fprintf(stderr, "zone send action must be specified\n");
+		err = -EINVAL;
+		goto close_fd;
+	}
+
+	if (cfg.zsa == NVME_ZNS_ZSA_SET_DESC_EXT){
+		if(!cfg.data_len){
+			cfg.data_len = get_zdes_bytes(fd, cfg.namespace_id);
+			if (cfg.data_len == 0){
+				fprintf(stderr, 
+				"Zone Descriptor Extensions are not supported\n");
+				goto close_fd;
+			}
+			else if (cfg.data_len < 0) {
+				fprintf(stderr, "Failed to retrive valid zdes\n");
```

Comment:
Failed to retrieve the Zone Description Zone Description Extension Size? (if it is being written out in the previous error message, I think the same should be done here so it is consistent.